addons/udes_stock/models/stock_move.py

```diff
@@ -310,13 +321,64 @@ def _create_moves_for_push(self, push, move_lines):
             created_moves |= move.copy(move_vals)
         return created_moves
 
+    def group_by_key(self):
+        # force recompute on u_grouping_key so we have an up-to-date key:
+        self._fields['u_grouping_key'].compute_value(self)
+
+        if any(pt.u_move_key_format is False
+               for pt in self.mapped('picking_id.picking_type_id')):
+            raise UserError(_("Cannot group move lines when their picking type"
+                              "has no grouping key set."))
+
+        return self.groupby(lambda ml: ml.u_grouping_key)
+
+    def refactor_action_group_by_move_key(self):
+        """
+        group the moves by the splitting criteria
+        for each resulting group of stock.moves:
+            create a new picking
+            split any stock.move records that are only partially covered by the
```

Comment:
Moves are not going to be split as part of this. Looks like you removed the one from the move line method

Reply:
I thought I had already done that :confused: